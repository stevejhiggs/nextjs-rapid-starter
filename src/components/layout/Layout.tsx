import React from 'react';

import { Grid, GridItem } from '@chakra-ui/react';
import Header from './Header';
import Nav from './Nav';
import Footer from './Footer';

interface Properties {
  children: React.ReactNode;
}

const Layout: React.FC<Properties> = ({ children }) => {
  return (
    <Grid templateColumns="200px 1fr" templateRows="auto 1fr auto" gap="10px" height="100vh">
      <GridItem as="header" gridArea="1 / 1 / 2 / 6">
        <Header />
      </GridItem>

      <GridItem as="nav" gridArea="2 / 1 / 5 / 2">
        <Nav />
      </GridItem>

      <GridItem as="main" gridArea="2 / 2 / 5 / 6">
        {children}
      </GridItem>

      <GridItem as="footer" gridArea="5 / 1 / 6 / 6">
        <Footer />
      </GridItem>
    </Grid>
  );
};

export default Layout;
