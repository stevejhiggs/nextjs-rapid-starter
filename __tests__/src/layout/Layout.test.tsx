import React from 'react';
import { cleanup, render } from '@testing-library/react';
import Layout from '@/src/layout/Layout';

afterEach(cleanup);

describe('Components -> Layout', () => {
  it('should match the snapshot', () => {
    const { container } = render(<Layout />);
    expect(container).toMatchSnapshot();
  });
});
