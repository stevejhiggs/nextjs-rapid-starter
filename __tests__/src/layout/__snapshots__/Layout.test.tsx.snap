// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Components -> Layout should match the snapshot 1`] = `
<div>
  <div
    class="css-1ewyn93"
  >
    <header
      class="css-1p4uj4t"
    >
      <nav
        class="css-1hn0bm1"
      >
        <div
          class="css-1c714fi"
        >
          <h1
            class="chakra-heading css-0"
            data-testid="page-heading-title"
          >
            Next.js starter
          </h1>
        </div>
        <div
          class="css-m9ssb"
        >
          <svg
            fill="white"
            viewBox="0 0 20 20"
            width="12px"
            xmlns="http://www.w3.org/2000/svg"
          >
            <title>
              Menu
            </title>
            <path
              d="M0 3h20v2H0V3zm0 6h20v2H0V9zm0 6h20v2H0v-2z"
            />
          </svg>
        </div>
        <div
          class="css-t2kg2r"
        >
          <p
            class="chakra-text css-dtinqn"
          >
            Menu1
          </p>
          <p
            class="chakra-text css-dtinqn"
          >
            Menu2
          </p>
          <p
            class="chakra-text css-dtinqn"
          >
            Menu3
          </p>
        </div>
        <div
          class="css-mg6i13"
        >
          <button
            class="chakra-button css-1tvtpuq"
            type="button"
          >
            Example Button
          </button>
        </div>
      </nav>
    </header>
    <nav
      class="css-19je435"
    >
      <ul
        class="css-0"
      >
        <li
          class="css-pmguu6"
        >
          <a
            class="chakra-link css-0"
            href="/"
          >
            Home
          </a>
        </li>
        <li
          class="css-pmguu6"
        >
          <a
            class="chakra-link css-0"
            href="/about"
          >
            About
          </a>
        </li>
        <li
          class="css-pmguu6"
        >
          <a
            class="chakra-link css-0"
            href="/customComponents"
          >
            Custom components
          </a>
        </li>
        <li
          class="css-pmguu6"
        >
          <a
            class="chakra-link css-0"
            href="/serverRendered"
          >
            Server rendered
          </a>
        </li>
        <li
          class="css-pmguu6"
        >
          <a
            class="chakra-link css-0"
            href="/staticallyGenerated"
          >
            Statically generated
          </a>
        </li>
        <li
          class="css-pmguu6"
        >
          <a
            class="chakra-link css-0"
            href="/showRouter/blahblahMadeUpValue"
          >
            Dynamic route
          </a>
        </li>
        <li
          class="css-pmguu6"
        >
          <a
            class="chakra-link css-0"
            href="/i18n/staticPage"
          >
            i18n - Static
          </a>
        </li>
        <li
          class="css-pmguu6"
        >
          <a
            class="chakra-link css-0"
            href="/i18n/serverRendered"
          >
            i18n - Server rendered
          </a>
        </li>
        <li
          class="css-pmguu6"
        >
          <a
            class="chakra-link css-0"
            href="/getStaticPaths/blogArticleA"
          >
            GetStaticPaths
          </a>
        </li>
        <li
          class="css-pmguu6"
        >
          <a
            class="chakra-link css-0"
            href="/getStaticPathsChildren/blog/articles/A"
          >
            GetStaticPaths - Nested
          </a>
        </li>
      </ul>
    </nav>
    <main
      class="css-92di6r"
    />
    <footer
      class="css-12zq1je"
    />
  </div>
</div>
`;
